packages/runtime-core/src/apiCreateApp.ts

```diff
@@ -125,8 +125,13 @@ export function createAppAPI<HostElement>(
   hydrate?: RootHydrateFunction
 ): CreateAppFunction<HostElement> {
   return function createApp(rootComponent, rootProps = null) {
+    if (!isObject(rootComponent)) {
```

Comment:
Functional components can also be used as root components

```suggestion
    if (!isObject(rootComponent) && !isFunction(rootComponent)) {
```

Reply:
This check should be dev only as it is likely an error on the user behalf and they will detect it before deploying. That way the code is removed from production too